src/Application/Actions/Donations/Confirm.php

```diff
@@ -116,17 +118,16 @@ protected function action(Request $request, Response $response, array $args): Re
             throw new HttpBadRequestException($request, $message);
         }
 
-        $paymentIntentId = $donation->getTransactionId();
         try {
-            $paymentMethod = $this->stripe->updatePaymentMethodBillingDetail($paymentMethodId, $donation);
```

Comment:
this is a slight performance degradation - not using the paymentMethod returned from `updatePaymentMethodBillingDetail` means we have to retrieve it from Stripe later.  But applying command query separation simplifies moving code around.